ios/Classes/PurchasesFlutterPlugin.m

```diff
@@ -479,10 +492,23 @@ - (void)closeWithResult:(FlutterResult)result {
 #pragma mark Delegate Methods
 
 - (void)purchases:(RCPurchases *)purchases didReceiveUpdatedPurchaserInfo:(RCPurchaserInfo *)purchaserInfo {
-    [self.channel invokeMethod:RNPurchasesPurchaserInfoUpdatedEvent
+    [self.channel invokeMethod:PurchasesPurchaserInfoUpdatedEvent
                      arguments:purchaserInfo.dictionary];
 }
 
+-(void) purchases:(RCPurchases *)purchases shouldPurchasePromoProduct:(SKProduct *)product defermentBlock:(RCDeferredPromotionalPurchaseBlock)makeDeferredPurchase {
+    if (!self.startPurchaseBlocks) {
+            self.startPurchaseBlocks = [NSMutableArray array];
+    }
+
+    [self.startPurchaseBlocks addObject:makeDeferredPurchase];
+    NSInteger position = [self.startPurchaseBlocks count] - 1;
+    [self.channel invokeMethod:PurchasesReadyForPromotedProductPurchaseEvent
+                     arguments:@{@"callbackID": [NSNumber numberWithInteger:position],
```

Comment:
I think you can replace the `[NSNumber numberWithInteger:position]` with simply `@(position)`